src/library/dlgtrackinfo.h

```diff
@@ -18,22 +20,22 @@
 class DlgTrackInfo : public QDialog, public Ui::DlgTrackInfo {
     Q_OBJECT
   public:
-    DlgTrackInfo(UserSettingsPointer pConfig, QWidget* parent);
+    DlgTrackInfo(UserSettingsPointer pConfig, QWidget* parent, const TrackModel* trackModel = nullptr);
     virtual ~DlgTrackInfo();
 
   public slots:
     // Not thread safe. Only invoke via AutoConnection or QueuedConnection, not
     // directly!
     void loadTrack(TrackPointer pTrack);
+    void loadTrack(QModelIndex index);
 
   signals:
     void next();
     void previous();
-    void showTagFetcher(TrackPointer pTrack);
 
   private slots:
-    void slotNext();
-    void slotPrev();
+    void slotNext(bool loadTrackInTagFetcher = true);
```

Comment:
I'm not a fan of using default arguments in slots. This could cause unintended effects as we have seen with the workaround when forwarding another signal. Better use 2 separate slots with different names.

Reply:
Do we need this parameter at all? Shouldn't the track always be forwarded to m_pTagFetcher if it is not null?

Reply:
> I'm not a fan of using default arguments in slots. This could cause unintended effects as we have seen with the workaround when forwarding another signal. Better use 2 separate slots with different names.

This certainly would be the better way. We can have two separate slots for button and `m_pTagFetcher`.

Reply:
> Do we need this parameter at all? Shouldn't the track always be forwarded to m_pTagFetcher if it is not null?

If the signal is recieved from `m_pTagFetcher`, it means that the track has already changed in `m_pTagFetcher`, calling `m_pTagFetcher->loadTrack` will only cause a reload.

Reply:
Since both @Be and me stumbled upon this code the intended behavior doesn't seem to be obvious. If the slots are used for different use cases then it might be an option to separate them. Or rename them and their parameter, whatever.

Choosing a concrete solution that resolves the conflict is up to you. Simple rule: If you need to explain your code then it needs to be improved.